packages/cart/src/services/cart-module.ts

```diff
@@ -583,7 +587,13 @@ export default class CartModuleService implements ICartModuleService {
       )
     } else {
       const data = Array.isArray(cartIdOrData) ? cartIdOrData : [cartIdOrData]
-      methods = await this.addShippingMethodsBulk_(data, sharedContext)
+      methods = await this.addShippingMethodsBulk_(
+        data as WithRequiredProperty<
+          CartTypes.CreateShippingMethodDTO,
+          "cart_id"
```

Comment:
suggestion: isn't the cart_id already non optional in the interface?

Reply:
but the type `CartTypes.CreateShippingMethodDTO` already have the cart_id as non optional here. Or did i mixed the interfaces. The withRequired only remove the `?` from the prop

Reply:
cart_id  is already non optional in the interface below, so the usage of `WithRequiredProperty` wont do anything more that what it already is

<img width="372" alt="Screenshot 2024-01-18 at 14 40 39" src="https://github.com/medusajs/medusa/assets/25098370/cbd366ed-26d0-486f-9266-15f4fb412aff">

Reply:
though, line 576 it should be made optional